src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+
+  function averageMark({ grades }: Student): number {
+    return grades.reduce((acc, cur) => acc + cur, 0) / grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return (order === 'asc')
+        ? sortedStudents.sort(
+          (a: Student, b: Student) => a[sortBy].localeCompare(b[sortBy]),
+        )
+
+        : sortedStudents.sort(
+          (a: Student, b: Student) => b[sortBy].localeCompare(a[sortBy]),
+        );
+
+    case SortType.Age:
+      return (order === 'asc')
+        ? sortedStudents.sort((a: Student, b: Student) => a.age - b.age)
+        : sortedStudents.sort((a: Student, b: Student) => b.age - a.age);
+
+    case SortType.Married:
+      return (order === 'asc')
+        ? sortedStudents.sort(
+          (a: Student, b: Student) => Number(a.married) - Number(b.married),
+        )
+        : sortedStudents.sort(
+          (a: Student, b: Student) => Number(b.married) - Number(a.married),
+        );
```

Comment:
you can just group these cases
```suggestion
    case SortType.Age,
    case SortType.Married:
      return (order === 'asc')
        ? sortedStudents.sort((a, b) => Number(a.married) - Number(b.married))
        : sortedStudents.sort((a, b) => Number(b.married) - Number(a.married));
```